lib/python/picongpu/plugins/plot_mpl/transition_radiation_visualizer.py

```diff
@@ -0,0 +1,275 @@
+            self.ax.set_title(
+                "Angular transition radiation distribution for " + species)
+            self.ax.set_xlabel(r"Detector Angle $\theta$")
+            self.ax.set_ylabel(r"Spectral Power $d^2 W / d\omega d\Omega$")
+        elif self.type == "sliceoverphi":
+            self.ax.set_title(
+                "Angular transition radiation distribution for " + species)
+            self.ax.set_xlabel(r"Detector Angle $\phi$")
+            self.ax.set_ylabel(r"Spectral Power $d^2 W / d\omega d\Omega$")
+        elif self.type == "heatmap":
+            self.ax.set_title(
+                "Angular transition radiation distribution for " + species)
+            self.ax.set_xlabel(r"Detector Angle $\theta$")
+            self.ax.set_ylabel(r"Detector Angle $\phi$")
+
+
+if __name__ == "__main__":
+    def main():
+        import sys
+        import getopt
+        import matplotlib.pyplot as plt
+
+        def usage():
+            print("usage:")
+            print("python", sys.argv[0], "\n",
+                  "\t-p, --path\t<path to run_directory>\n"
+                  "\t-i, --iteration\t<iteration>\n"
+                  "\t-s, --species\t<particle species>\n"
+                  "\t-t, --type\t<figure type>\n"
+                  "\t\tavailable figure types:\n"
+                  "\t\t -'spectrum'\n"
+                  "\t\t -'sliceovertheta'\n"
+                  "\t\t -'sliceoverphi'\n"
+                  "\t\t -'heatmap'\n"
+                  "\t-P, --phi\t<index of polar angle phi>\n"
+                  "\t-T, --theta\t<index of azimuth angle theta>\n"
+                  "\t-O, --omega\t<index of frequency omega>")
+
+        path = None
+        iteration = None
+        species = None
+        type = None
+
+        # Values for a plot with a fixed angle or frequency
+        phi = None
+        theta = None
+        omega = None
+
+        try:
+            opts, args = getopt.getopt(sys.argv[1:], "hp:i:s:t:P:T:O:",
+                                       ["help", "path", "iteration",
+                                        "species", "type", "phi",
+                                        "theta", "omega"])
+        except getopt.GetoptError as err:
+            print(err)
+            usage()
+            sys.exit(2)
+
+        # go through kwargs
+        for opt, arg in opts:
+            if opt in ["-h", "--help"]:
+                usage()
+                sys.exit()
+            elif opt in ["-p", "--path"]:
+                path = arg
+            elif opt in ["-i", "--iteration"]:
+                iteration = int(arg)
+            elif opt in ["-s", "--species"]:
+                species = arg
+            elif opt in ["-t", "--type"]:
+                type = arg
+            elif opt in ["-P", "--phi"]:
+                phi = arg
+            elif opt in ["-T", "--theta"]:
+                theta = arg
+            elif opt in ["-O", "--omega"]:
+                omega = arg
```

Comment:
In order to avoid this manual parsing of arguments and the manual help page, I think using the module `argparse`, see e.g. [radiation plotting here](https://github.com/ComputationalRadiationPhysics/picongpu/blob/d340812641320d3a61e343a521287fd0720ba55b/src/tools/bin/plotRadiation#L67-L210)

Reply:
I followed the [Energy histogram visualizer](https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/lib/python/picongpu/plugins/plot_mpl/energy_histogram_visualizer.py) with the argument parsing for consistency...